src/fsharp/tast.fs

```diff
@@ -1912,8 +1912,8 @@ and
 
     /// Indicates if a type variable has been solved.
     member x.IsSolved = 
-        match x.Solution with 
-        | None -> false
+        match box x.Data.typar_solution with 
```

Comment:
Wonder if there is an opportunity to cleanup the match on this which are repeated.

Not that match is incorrect, but `Data.HasTypeArgumentSolution` would probably read better and not leak underlying representation detail with the overall logic (even if this is member implementation and not higher level code flow).

The only legit match seems to be the one actually matching with `TType_measure unt`.